lib/dialects/mssql/query.js

```diff
@@ -164,8 +164,8 @@ module.exports = (function() {
   Query.prototype.handleShowTablesQuery = function(results) {
     return results.map(function(resultSet) {
       return {
-        tableName: resultSet.TABLE_NAME,
-        schema: resultSet.TABLE_SCHEMA
+        tableName: resultSet.tableName,
+        schema: resultSet.schema
```

Comment:
if you take a look at the abstract query way of handling this over [here](https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query.js#L134), you'll see that it by default flattens the result set. Therefore the query above, whether it was modified or not, would always end up returning a result set that looks like ['myTable', 'dbo', etc] rather than [{tableName: 'myTable', schema:'dbo'}, etc]. I believe the existing code is correct, and @Groxot should describe the error that he's facing